private/tst/avb_streamhandler/src/IasTestSystemdWatchdog.cpp

```diff
@@ -0,0 +1,233 @@
+    ASSERT_TRUE(mWatchdogTimerRegistration != nullptr);
+
+  }
+
+  virtual void TearDown()
+  {
+    delete mWdManager;
+    mWatchdogTimerRegistration = nullptr;
+  }
+
+  DltContext mDltContext;
+  std::shared_ptr<IasWatchdogTimerRegistration> mWatchdogTimerRegistration;
+  IasSystemdWatchdogManager *mWdManager;
+};
+
+TEST_F(IasTestSystemdWatchdog, init)
+{
+  ASSERT_EQ(setenv("WATCHDOG_USEC", "30000000", 1), 0);
+  IasWatchdogResult result = mWdManager->init(mWatchdogTimerRegistration);
+  ASSERT_EQ(result, IasMediaTransportAvb::IasResult::cOk);
+
+  ASSERT_EQ(setenv("WATCHDOG_USEC", "99999", 1), 0);
+  result = mWdManager->init(mWatchdogTimerRegistration);
+  ASSERT_EQ(result, IasMediaTransportAvb::IasResult::cInitFailed);
+
+  result = mWdManager->init(nullptr);
+  ASSERT_EQ(result, IasMediaTransportAvb::IasResult::cParameterInvalid);
+
+  ASSERT_EQ(unsetenv("WATCHDOG_USEC"), 0);
+  result = mWdManager->init(mWatchdogTimerRegistration);
+  ASSERT_EQ(result, IasMediaTransportAvb::IasResult::cInitFailed);
+}
+
+TEST_F(IasTestSystemdWatchdog, createWatchdog)
+{
+  IasWatchdogInterface *wdInterface = nullptr;
+  wdInterface = mWdManager->createWatchdog();
+  ASSERT_NE(wdInterface, nullptr);
+  ASSERT_EQ(mWdManager->mWatchdogInterfaces.size(), 1);
+
+  wdInterface = nullptr;
+  wdInterface = mWdManager->createWatchdog(30000, "Malekith");
+  ASSERT_NE(wdInterface, nullptr);
+  ASSERT_EQ(mWdManager->mWatchdogInterfaces.size(), 2);
+}
+
+TEST_F(IasTestSystemdWatchdog, destroyWatchdog)
+{
+  IasSystemdWatchdogManager *altWdManager = nullptr;
+  altWdManager = new (std::nothrow) IasSystemdWatchdogManager(mDltContext);
+  IasWatchdogInterface *wdIf_1, *wdIf_2;
+  IasWatchdogResult res = IasMediaTransportAvb::IasResult::cOk;
+  wdIf_1 = wdIf_2 = nullptr;
+
+  wdIf_1 = mWdManager->createWatchdog(30000, "Marvel");
+  ASSERT_NE(wdIf_1, nullptr);
+
+  wdIf_2 = altWdManager->createWatchdog(30000, "DC");
+  ASSERT_NE(wdIf_2, nullptr);
+
+  res = mWdManager->destroyWatchdog(nullptr);
+  ASSERT_EQ(res, IasMediaTransportAvb::IasResult::cParameterInvalid);
+
+  res = mWdManager->destroyWatchdog(wdIf_2);
+  ASSERT_EQ(res, IasMediaTransportAvb::IasResult::cObjectNotFound);
+
+  res = mWdManager->destroyWatchdog(wdIf_1);
+  ASSERT_EQ(res, IasMediaTransportAvb::IasResult::cOk);
+
+  res = altWdManager->destroyWatchdog(wdIf_2);
+  ASSERT_EQ(res, IasMediaTransportAvb::IasResult::cOk);
+}
+
+TEST_F(IasTestSystemdWatchdog, removeWatchdog)
+{
+  IasWatchdogInterface *wdIf = nullptr;
+  IasWatchdogResult res = IasMediaTransportAvb::IasResult::cOk;
+
+  wdIf = mWdManager->createWatchdog(30000, "Ultron");
+  ASSERT_NE(wdIf, nullptr);
+
+  res = mWdManager->removeWatchdog(wdIf);
+  ASSERT_EQ(res, IasMediaTransportAvb::IasResult::cOk);
+
+  res = mWdManager->removeWatchdog(wdIf);
+  ASSERT_EQ(res, IasMediaTransportAvb::IasResult::cObjectNotFound);
+
+  res = mWdManager->removeWatchdog(nullptr);
+  ASSERT_EQ(res, IasMediaTransportAvb::IasResult::cParameterInvalid);
```

Comment:
Note that all assertions for the `removeWatchdog` results could be `EXPECT_XX`. The one that creates the watchdog should remain an assertion, though.